### DISCUSSION TITLE
Equivalent for 'useRecoilTransactionObserver_UNSTABLE'

### DISCUSSION THREAD

**Discussion by akshayashok95** [pmndrs/jotai#870] [Q&A] [answered]
What's the equivalent for 'useRecoilTransactionObserver_UNSTABLE'in jotai ?

**Comment by dai-shi** [comment, marked as answer]
Would be able to explain the use case with a small code example?

**Reply by akshayashok95** [reply]
I was trying to persist the values using the snapshot provided by `useRecoilTransactionObserver_UNSTABLE` but i found this to be 
 helping the usecase https://jotai.org/docs/guides/persistence#a-simple-pattern-with-local-storage .

I have one question though , is there a easier way to do this for multiple atoms 
```
const strAtom = atom(localStorage.getItem('myKey') ?? 'foo')

const strAtomWithPersistence = atom(
  (get) => get(strAtom),
  (get, set, newStr) => {
    set(strAtom, newStr)
    localStorage.setItem('myKey', newStr)
  }
)
```

**Reply by dai-shi** [reply]
For multiple atoms with different keys? You could create a helper function.
```js
const createAtomWithPersistence = (key, defaultValue) => {
  const baseAtom = atom(localStorage.getItem(key) ?? defaultValue)
  const derivedAtom = atom(
    (get) => get(baseAtom),
    (get, set, newValue) => {
      set(baseAtom, newValue)
      localStorage.setItem(key, newValue)
    }
  )
  return derivedAtom
}
```

Or, do you mean something else?

**Reply by akshayashok95** [reply]
I'm sorry for not being clear . This is my use case
```
const persistAtom= useCallback(({snapshot})=>{
    localStorage.setItem({
      atom1:snapshot.getLoadable(atom1).contents,
      atom2:snapshot.getLoadable(atom2).contents,
    })
  })

  useRecoilTransactionObserver_UNSTABLE(persistAtom)
```

**Reply by dai-shi** [reply]
The easiest approach with that mental model is useEffect. (It's slightly delayed and has slight overhead, but should be reasonable in practice.)

```js
const EffectOnlyComponentJustUnderRoot = () => {
  const [values] = useAtom(useMemo(() => atom((get) => ({
    atom1: get(atom1),
    atom2: get(atom2),
  })), []))
  useEffect(() => {
    localStorage.setItem('key', values)
  }, [values])
  return null
}
```

The pure jotai way would like to avoid useEffect. Have you seen `atomWithStorage` in `jotai/utils`?

```js
import { atomWithStorage } from 'jotai/utils'
import { focusAtom } from 'jotai/optics'

const baseAtom = atomWithStorage('key', { atom1: '', atom2: '' })
const atom1 = focusAtom(baseAtom, optic => optic.prop('atom1'))
const atom2 = focusAtom(baseAtom, optic => optic.prop('atom2'))
```

(We could avoid `focusAtom` and hand build atoms.)

**Reply by akshayashok95** [reply]
Thanks for the quick response. This is really helpful . Keep up the awesome work
